storage/objects.go

```diff
@@ -19,6 +19,86 @@ type ObjectsClient struct {
 	client *client.Client
 }
 
+// GetObjectInput represents parameters to a GetObject operation.
+type GetInfoInput struct {
+	ObjectPath string
+	Headers    map[string]string
+}
+
+// GetObjectOutput contains the outputs for a GetObject operation. It is your
+// responsibility to ensure that the io.ReadCloser ObjectReader is closed.
+type GetInfoOutput struct {
+	ContentLength uint64
+	ContentType   string
+	LastModified  time.Time
+	ContentMD5    string
+	ETag          string
+	Metadata      map[string]string
+}
+
+// GetInfo sends a HEAD request to an object in the Manta service. This function
+// does not return a response body.
+func (s *ObjectsClient) GetInfo(ctx context.Context, input *GetInfoInput) (*GetInfoOutput, error) {
+	fullPath := path.Clean(path.Join("/", s.client.AccountName, input.ObjectPath))
+
+	headers := &http.Header{}
+	for key, value := range input.Headers {
+		headers.Set(key, value)
```

Comment:
Use `Add` instead of `Set` in order to allow multiple headers with the same name.  We should probably make a sweep for that exact usage elsewhere in the repo.